### PR TITLE
Add manual trigger feature for khchecks by creating kuberhealthy jobs crd

### PR DESCRIPTION
**Pull request by joshulyne**
#528 Manual trigger functionality using kuberhealthyjobs

### CODE REVIEW

File: pkg/checks/external/main.go [hunk 1]
```diff
@@ -124,14 +127,29 @@ func init() {
 }
 
 // New creates a new external checker
-func New(client *kubernetes.Clientset, checkConfig *khcheckcrd.KuberhealthyCheck, khCheckClient *khcheckcrd.KuberhealthyCheckClient, khStateClient *khstatecrd.KuberhealthyStateClient, reportingURL string) *Checker {
+func New(client *kubernetes.Clientset, checkConfig *khcheckcrd.KuberhealthyCheck, jobConfig *khjobcrd.KuberhealthyJob, khCheckClient *khcheckcrd.KuberhealthyCheckClient, khStateClient *khstatecrd.KuberhealthyStateClient, khJobClient *khjobcrd.KhjobV1Client, reportingURL string, khjob bool) *Checker {
```
Comment (integrii) [inline pkg/checks/external/main.go:130]: Consider refactor to `NewChecker` and `NewJob` funcs.   This lets us remove the `khjob bool` from this function signature.  New checks will continue to call `New` and `New` will simply call `NewChecker`.  Likewise, `NewJob` will create a new job.

File: cmd/kuberhealthy/kuberhealthy.go [hunk 2]
```diff
@@ -1319,32 +1318,28 @@ func (k *Kuberhealthy) externalCheckReportHandler(w http.ResponseWriter, r *http
...
-	jobDetails := k.stateReflector.CurrentStatus().JobDetails
+	khWorkload := determineKHWorkload(ipReport.Name, ipReport.Namespace)
+	log.Debugln("kh workload:", khWorkload)
+
 	checkRunDuration := time.Duration(0).String()
-	var khWorkload health.KHWorkload
-	if checkDetails != nil {
-		checkRunDuration = checkDetails[ipReport.Namespace+"/"+ipReport.Name].RunDuration
-		khWorkload = checkDetails[ipReport.Namespace+"/"+ipReport.Name].KHWorkload
+	if khWorkload == health.KHCheck {
```
Comment (joshulyne) [inline cmd/kuberhealthy/kuberhealthy.go:1325]: test for empty khworkload -- or use switch case

Comment (joshulyne) [reply cmd/kuberhealthy/kuberhealthy.go:1325]: return an error up

File: cmd/kuberhealthy/kuberhealthy.go [hunk 3]
```diff
@@ -1358,6 +1353,35 @@ func (k *Kuberhealthy) externalCheckReportHandler(w http.ResponseWriter, r *http
 	return nil
 }
 
+func determineKHWorkload(name string, namespace string) health.KHWorkload {
```
Comment (joshulyne) [inline cmd/kuberhealthy/kuberhealthy.go:1356]: comment on this function

File: cmd/kuberhealthy/Makefile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 IMAGE="kuberhealthy/kuberhealthy"
-TAG="v2.3.2"
+TAG="v2.4.0"
```
Comment (integrii) [inline cmd/kuberhealthy/Makefile:2]:
```suggestion
TAG="latest"
```

Let's move this to `latest` and only tag images with release versions when we do a release.  In the future, we will just make new builds of kuberhealthy use a commit hash until a release is cut.

**Review by integrii — CHANGES_REQUESTED** [review]

**Review by integrii — APPROVED** [review]
LGTM.  Feel free to merge after enough in-cluster testing looks good.  👍 🥳

**Pull request merged**